pkg/pgcapture/debounce.go

```diff
@@ -113,9 +113,18 @@ func (b *DebounceHandler) Handle(fn ModelAsyncHandlerFunc, checkpoint cursor.Che
 		}
 		key := debounceKey(change.New)
 		if prev, ok := b.store[key]; ok {
-			b.source.Commit(prev.Checkpoint)
+			// since requeue order is not guaranteed, we need to check if the new event is newer than the previous one
+			// then we commit the previous one and store the new one
+			// workaround for the LSN == 0 issue because schedule dump lsn is 0 and should be always the latest event
+			if change.Checkpoint.LSN == 0 || change.Checkpoint.After(prev.Checkpoint) {
```

Comment:
What will happen if change.Checkpoint == prev.Checkpoint?

Reply:
How about just handling and deleting the previous event? 
Since Pulsar would send the same message twice, otherwise we need to extract the message ID from the checkpoint. I think it's not convenient and doesn't happen often.